### DISCUSSION TITLE
Release 3.9

### DISCUSSION THREAD

**Discussion by revoxhere** [revoxhere/duino-coin#1705] [General]
## Hello miners! 
As you can see, this is the last update before we drop a major 4.0 release. This is also the last release with separate ESP8266 and ESP32 codes that require you to install old library versions. 
Theoretically could've just skipped this one, but there have been quite a few changes (eg. fixing compilation errors on Linux systems) that had to be adressed. 
## What's new, what's changed...
* Fixed a bug with Arduino Code on *nix systems by @trazfr in https://github.com/revoxhere/duino-coin/pull/1677
* Fixed a bug with URL encoding in the PC Miner by @matteocrippa in https://github.com/revoxhere/duino-coin/pull/1683
* Fixed a bug with auto-updater by @umikali222 in https://github.com/revoxhere/duino-coin/pull/1649
* Added Vietnamese translation by @MinatoIsuki in https://github.com/revoxhere/duino-coin/pull/1650
* Added policy links to README by @Tech1k in https://github.com/revoxhere/duino-coin/pull/1663
* Updated ToS and privacy policy by @EinWildesPanda in https://github.com/revoxhere/duino-coin/pull/1647
* Updated german translations by @EinWildesPanda in https://github.com/revoxhere/duino-coin/pull/1648
* Updated Vietnamese translation by @MinatoIsuki in https://github.com/revoxhere/duino-coin/pull/1654
* Updated French README by @TheFrenchTechMan in https://github.com/revoxhere/duino-coin/pull/1652
* Updated Portuguese translation by @fatorius in https://github.com/revoxhere/duino-coin/pull/1668
* Updated project maintainers info by @Tech1k in https://github.com/revoxhere/duino-coin/pull/1687
* Updated README.md by @revoxhere in https://github.com/revoxhere/duino-coin/pull/1704

## New contributors - hello! 👋 
* @umikali222 made their first contribution in https://github.com/revoxhere/duino-coin/pull/1649
* @MinatoIsuki made their first contribution in https://github.com/revoxhere/duino-coin/pull/1650
* @fatorius made their first contribution in https://github.com/revoxhere/duino-coin/pull/1668
* @trazfr made their first contribution in https://github.com/revoxhere/duino-coin/pull/1677
* @matteocrippa made their first contribution in https://github.com/revoxhere/duino-coin/pull/1683

### ⚠️ NOTICE: .exe binaries may be detected as a virus by your antivirus software. This is a **false positive** caused by pyinstallers bootloader, read about it [here](https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg). Duino-Coin is not a virus.

**Full Changelog**: https://github.com/revoxhere/duino-coin/compare/3.5...3.9
As always, please report any encountered issues. Happy mining!

<hr /><em>This discussion was created from the release <a href='https://github.com/revoxhere/duino-coin/releases/tag/3.9'>Release 3.9</a>.</em>

**Comment by ratotimo** [comment]
Hello...
Friend, ask me a question?
I bought 2 Esp32 cards from China, their maximum hashrate is 21.14, I had seen that on average the hashrate was 40, I have already tested several configurations and I was unable to do so... Thank you in advance...
